npm/test/basics.js

```diff
@@ -496,12 +495,49 @@ test("language service diagnostics - web worker", async () => {
         let m1 = M(q1);
         return [m1];
     }
-}`,
-    true // PackageType "exe"
+}`
   );
   languageService.terminate();
   assert(gotDiagnostics);
 });
+
+test("language service configuration update", async () => {
+  const languageService = getLanguageServiceWorker();
+  let gotDiagnostics = false;
+  let expectedMessages = [
+    "entry point not found\n\nhelp: a single callable with the `@EntryPoint()` attribute must be present if no entry expression is provided",
+  ];
+  languageService.addEventListener("diagnostics", (event) => {
+    gotDiagnostics = true;
+    assert.equal(event.type, "diagnostics");
+    assert.equal(event.detail.diagnostics.length, expectedMessages.length);
+    event.detail.diagnostics.map((d, i) =>
+      assert.equal(d.message, expectedMessages[i])
+    );
+  });
+  await languageService.updateDocument(
+    "test.qs",
+    1,
+    `namespace Sample {
+    operation main() : Unit {
+    }
+}`
+  );
+  // Above document should have generated a missing entrypoint error
+  assert(gotDiagnostics);
+
+  // Reset expectations
+  gotDiagnostics = false;
+  expectedMessages = [];
+
+  await languageService.updateConfiguration({ packageType: "lib" });
+
+  // Updating the config should cause another diagnostics event clearing the errors
```

Comment:
Should you be verifying the expectedMessages is emtpy here?

Reply:
It's done in the callback above. (I always find it a challenge to test callbacks, if you have any refactoring tricks I'd welcome them).